hs_core/hydroshare/resource.py

```diff
@@ -196,27 +196,47 @@ def get_checksum(pk):
 def check_resource_files(files=()):
     """
     internally used method to check whether the uploaded files are within
-    the supported maximal size limit
+    the supported maximal size limit. Also returns sum size of all files for
+    quota check purpose if all files are within allowed size limit
 
     Parameters:
     files - list of Django File or UploadedFile objects to be attached to the resource
-    Returns:    True if files are supported; otherwise, returns False
+    Returns: (status, sum_size) tuple where status is True if files are within FILE_SIZE_LIMIT
+             and False if not, and sum_size is the size summation over all files if status is
+             True, and -1 if status is False
     """
+    sum = 0
     for file in files:
         if not isinstance(file, UploadedFile):
             # if file is already on the server, e.g., a file transferred directly from iRODS,
```

Comment:
@alvacouch I am not sure if I understand your question correctly, so let me explain what it does here to clear any confusion first. This function checks each file among those to be added when creating a new resource or adding files to an existing resource, to make sure each uploaded file is within 1GB size limit, and if any uploaded file is larger than 1GB size limit, this function returns false. Since files coming from iRODS are not subject to size limitation, these files' sizes are only counted toward the total size of the files being added to a resource, but not subject to 1GB size limit check. The returned total sum size of all files that are to be added to a resource will be used to add on top of the used quota of this resource's quota holder, and if it is over the quota hard limit, this action will not be allowed. Not sure whether this clarifies and answers your question, but let me know if it does not so we can keep the discussion going.